database_cleanup/tests/test_database_cleanup.py

```diff
@@ -34,16 +48,18 @@ def test_database_cleanup(self):
             self.env.ref('database_cleanup.test_no_data_entry')
 
         # create a nonexistent model
-        self.env['ir.model'].create({
+        self.model = self.env['ir.model'].create({
             'name': 'Database cleanup test model',
             'model': 'x_database.cleanup.test.model',
         })
+
         self.env.cr.execute(
             'insert into ir_attachment (name, res_model, res_id, type) values '
             "('test attachment', 'database.cleanup.test.model', 42, 'binary')")
         self.registry.models.pop('x_database.cleanup.test.model')
         self.registry._pure_function_fields.pop(
             'x_database.cleanup.test.model')
+        self.registry.setup_models(self.env.cr, partial=False)
```

Comment:
right, doing this actually makes this a good model, so the next line fails because there's nothing to purge. Just pop the model from the registry immediately, then this goes well